### PR TITLE
devtools: fold local review experiment into planner

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary

- Remove the obsolete experimental ai/local-review-workflow.sh path.
- Fold useful review checks into devtools/local-validation-plan.sh: advisory
  raw allocation scan, changed-test antipattern scan, and mock distcheck for
  distribution-risk test/build changes.
- Make the planner derive its default base from the worktree creation baseline
  and use that base for local Cubic review.
- Move the useful prompt-audit guidance into the maintained local container
  testing skill without launching another local AI CLI.
- Keep shared prompt assets under ai/ and document that directory as the
  central prompt-asset library.
- Remove stale references to the deleted workflow paths.

## Validation

- sh -n devtools/local-validation-plan.sh
- devtools/local-validation-plan.sh --base main
- git diff --cached --check
- stale-reference search for local-review-workflow, .agent/workflows, /audit,
  and formerly described: no matches

## Notes

This intentionally keeps the old experiment out of the active tree instead of
carrying a second local review path. The planner now compares committed local
work against the worktree baseline by default, so local Cubic is not affected
when origin/main advances after the worktree was created.

### CODE REVIEW

**Comment by rgerhards** [comment]
Follow-up pushed in 07565776f. This fixes the valid planner robustness findings identified by cubic: deleted tests are no longer fed into focused CI_MAKE_CHECK_TESTS, modified distribution-risk files now trigger mock distcheck, name-status parsing uses tab fields, untracked C/H raw-allocation scan emits a path header, and +++ parsing no longer splits paths on whitespace. It also folds the old .agent/workflows/audit.md prompt workflow into the maintained local container testing skill as a late prompt-audit stage, without launching local AI CLIs, and reorders local validation so Ubuntu 26.04 run-ci executes before the slower static analyzer.\n\nValidation run locally: sh -n devtools/local-validation-plan.sh; git diff --check main..HEAD; devtools/local-validation-plan.sh --base main --run. I also dry-ran a throwaway worktree edge case showing modified tests/Makefile.am now recommends mock distcheck. A broader accidental --run against the worktree-creation baseline included upstream main changes after rebase and hit unrelated imdtls-sessionbreak.sh locally, so the final PR-scoped validation used --base main.

File: devtools/local-validation-plan.sh [hunk 1]
```diff
@@ -273,6 +341,63 @@ run_format_code_check() {
...
+				if (line ~ /(^|[^A-Za-z0-9_])(malloc|calloc|realloc|strdup|strndup)[[:space:]]*\(/ &&
+				    line !~ /(^|[^A-Za-z0-9_])CHK(malloc|realloc)[[:space:]]*\(/) {
+					if (file == "") {
+						file = "(untracked C/H file)"
+					}
+					print file ": " line
+				}
+			}
+		'
+	)"
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply devtools/local-validation-plan.sh:382]: Done in ed91aebc4. The scanner now emits a +++ b/path header for untracked C/H files and parses the path with substr(/bin/bash, 7), so untracked files and paths containing spaces are attributed correctly.

File: devtools/local-validation-plan.sh [hunk 2]
```diff
@@ -273,6 +341,63 @@ run_format_code_check() {
...
+		return 0
+	fi
+	if [ -x devtools/check-test-antipatterns.sh ]; then
+		# This helper is advisory and exits successfully even with findings.
+		# shellcheck disable=SC2086
+		devtools/check-test-antipatterns.sh $tests
+	else
+		echo "warning: devtools/check-test-antipatterns.sh missing or not executable; skipping test antipattern scan" >&2
+	fi
+}
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply devtools/local-validation-plan.sh:420]: Done in ed91aebc4. The antipattern scan no longer builds an unquoted space-separated test list; it iterates changed top-level tests and invokes the helper with each path quoted.

File: devtools/local-validation-plan.sh [hunk 3]
```diff
@@ -139,6 +178,21 @@ fi
...
+			if ($i ~ /^tests\/[^/]+\.sh$/ ||
+			    $i ~ /(^|\/)Makefile\.am$/ ||
+			    $i == "configure.ac" ||
+			    $i ~ /^m4\//) {
+				found = 1
+			}
+		}
+	}
+	END { exit found ? 0 : 1 }
+' "$tmp_status"; then
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply devtools/local-validation-plan.sh:194]: Done in ed91aebc4. The dist-risk awk parser now sets FS to tab and includes ACDMR status letters, so git diff --name-status paths with spaces remain single path fields.
